frontend/src/views/admin/matching/reducers.ts

```diff
@@ -154,4 +156,114 @@ const matchingDataReducer = createReducer(initialState, {
     [SET_UPDATED]: (state, action) => {
         return { ...state, updated: action.payload };
     },
+    [TOGGLE_STARRED]: (state, action) => {
+        // Check if a match with this applicant ID and position ID already exists
+        const existingMatch = state.matches.find(
+            (match) =>
+                match.utorid === action.payload.utorid &&
+                match.positionCode === action.payload.positionCode
+        );
+
+        // If a match doesn't already exist, this applicant is implied to not yet be starred:
+        if (!existingMatch) {
+            const newMatch: RawMatch = {
+                utorid: action.payload.utorid,
+                positionCode: action.payload.positionCode,
+                starred: true,
+            };
+
+            return {
+                ...state,
+                matches: [...state.matches, newMatch],
+                updated: true,
+            };
+        }
+
+        // Item exists, so we have to update it
+        return {
+            ...state,
+            matches: state.matches
+                .map((match) => {
```

Comment:
You can use `flatMap` for this and `return []` if you don't want to include an item.